src/posix/platform/netif.cpp

```diff
@@ -229,15 +527,35 @@ static void processStateChange(otChangedFlags aFlags, void *aContext)
 
 static void processReceive(otMessage *aMessage, void *aContext)
 {
-    char     packet[kMaxIp6Size];
-    otError  error  = OT_ERROR_NONE;
-    uint16_t length = otMessageGetLength(aMessage);
+    char     packet[kMaxIp6Size + 4];
+    otError  error      = OT_ERROR_NONE;
+    uint16_t length     = otMessageGetLength(aMessage);
+    size_t   offset     = 0;
+    uint16_t max_length = sizeof(packet) - 4;
+#if defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
+    offset += 4;
```

Comment:
See the man pages for tun/tap on BSDs.  The kernel/userland interface uses the four bytes to distinguish IPv6 from IPv4 -- while it's not technically necessary for it to do that (it could look at the packet), that's how the interface was originally developed and it's now there for legacy reasons.  wpantund has the same code.  I'll add a comment.